api/src/main/java/marquez/MarquezApp.java

```diff
@@ -154,6 +171,35 @@ public void run(@NonNull MarquezConfig config, @NonNull Environment env) {
     Exclusions.use(exclusions);
   }
 
+  private OpenSearchClient newOpenSearchClient(SearchConfig searchConfig) {
+    final HttpHost host =
+        new HttpHost(searchConfig.getHost(), searchConfig.getPort(), searchConfig.getScheme());
+    final BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+    credentialsProvider.setCredentials(
+        new AuthScope(host),
+        new UsernamePasswordCredentials(searchConfig.getUsername(), searchConfig.getPassword()));
+    final RestClient restClient =
+        RestClient.builder(host)
+            .setHttpClientConfigCallback(
+                httpClientBuilder ->
+                    httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider))
+            .build();
+
+    JacksonJsonpMapper jsonpMapper = new JacksonJsonpMapper();
```

Comment:
We may also want to consider `PoolingAsyncClientConnectionManager` support as outlined in the OpenSearch [docs](https://opensearch.org/docs/latest/clients/java/) for the `jvm` client. Not a requirement for this PR as this feature will be in `beta` and load testing will help us decide on whether pooling is needed.